usr/share/rear/prep/default/310_include_uefi_tools.sh

```diff
@@ -5,10 +5,13 @@ if grep -qw 'noefi' /proc/cmdline; then
     return
 fi
 
-# next step, is checking /boot/efi directory case-insensitive for the /EFI part (we need it)
-if [[ $(find /boot -maxdepth 1 -iname efi -type d | wc -l) -eq 0 ]] ; then
-    return    # must be mounted
-fi
+# Next step, is checking /boot/efi directory case-insensitive for the /EFI part (we need it).
+# If no /boot/[eE][fF][iI] directory can be found we cannot copy the UEFI binaries we might need.
```

Comment:
@schlomo Indeed `[[ -d /boot/[eE][fF][iI] ]]` is simpler.

["1. Identify the EFI System Partition" and "2. Determine its mount point" are already done in 320_include_uefi... starting from line 56](https://github.com/rear/rear/blob/19ec5cd3d5e69b99c76c7cb9d2437ff0278ef45b/usr/share/rear/prep/default/320_include_uefi_env.sh#L56) An export there make Rear ready for step number 3 "Use that instead of `/boot/efi`".

### Why check for /boot/efi before determining esp_mount_point?
For [320_include_uefi_env#Line50](https://github.com/rear/rear/blob/19ec5cd3d5e69b99c76c7cb9d2437ff0278ef45b/usr/share/rear/prep/default/320_include_uefi_env.sh#L50) I am still wondering why:

> ...(we need it)

? 
Best guess, to prevent an expensive `awk` lookup in `/proc/mounts`?


Note: there is no assumption that EFI boot is mounted on `/boot/[eE][fF][iI]`. The `-d` solution is an abstraction to not bother about details where exactly the ESP is mounted. The earlier code to check for `/boot/efi` breaks Arch Linux. That is why I suggested a case insensitive solution, just to not break Arch Linux. Despite the ESP being mount at `/boot` in Arch Linux with Gummiboot/systemd-boot, the result for `[[ -d /boot/[eE][fF][iI] ]]` is still positive, thus being able to detect the `esp_mount_point`, `UEFI_FS_TYPE` and `USING_UEFI_BOOTLOADER=1` [later on in 320_include_uefi_env.sh](https://github.com/rear/rear/blob/19ec5cd3d5e69b99c76c7cb9d2437ff0278ef45b/usr/share/rear/prep/default/320_include_uefi_env.sh#L60).

@jsmeix Some more background regarding ESP mount point locations:

> The ESP's mount point in Linux is largely irrelevant, at least on an abstract level. It does matter for some utilities, though -- for instance, rEFInd's installer looks for the ESP at /boot or /boot/efi, and won't work if it's mounted elsewhere. You've also got to know where it's mounted yourself, of course, and adjust any commands that reference files on the ESP. 

> There's one special case where the ESP's mount point is very important: If you mount the ESP at /boot, then normal procedures and tools that copy your kernel to /boot will end up copying your kernel to the ESP. This may be handy if you're using the EFI stub loader or some other tool that relies on the EFI being able to read the kernel (such as ELILO). This is why mounting the ESP at /boot is popular on Arch Linux; those users favor gummiboot, rEFInd, or a direct boot of the kernel (as you're doing), so putting the kernel on the ESP simplifies configuration. 
source: https://forums.gentoo.org/viewtopic-t-959396-start-0.html#7308500

Back to the problem: _the proper solution to handle the ESP_.

I don't think there is only one "proper" solution because the hard-coded `/boot/efi` serves multiple purposes. Sometimes to detect whether a system is EFI booted, other times to find "Boot loader" (files), in other cases to find "Boot manager" (files), and in other cases I can't figure out the intentions of the code.
Searching for [root efi](https://github.com/rear/rear/search?utf8=%E2%9C%93&q=boot+efi&type=Code)
 returns f.e. [510_selinux_fixfiles_exclude_dirs.sh](https://github.com/rear/rear/blob/2c12a52b3322f833c81cae2a8bc5baa799c3187d/usr/share/rear/restore/NETFS/Linux-i386/510_selinux_fixfiles_exclude_dirs.sh):
```
cat > $TARGET_FS_ROOT/etc/selinux/fixfiles_exclude_dirs <<EOF
/boot/efi
/boot/efi(/.*)?
EOF
```
For example for "Arch Linux" with SElinux enabled, the ESP is still mounted at `/boot`. Should in that case the output to exclude dirs become:
```
/boot/EFI
/boot/EFI(/.*)?
```
or the ESP mount point:
```
/boot
/boot(/.*)?
```
? I don't have a clue, because I never used SElinux.

@gdha When using the ESP mount point location, still doesn't solve the character case differences between different Linux distributions, for the (efi|EFI) named ESP subfolder:

| OS | ESP mount point | subfolder name | 
|---|---|---|
| Regular Linux | /boot/efi | efi | 
| Arch Linux | /boot | EFI |